homeassistant/components/imap/config_flow.py

```diff
@@ -128,3 +138,56 @@ async def async_step_reauth_confirm(
             ),
             errors=errors,
         )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> OptionsFlow:
+        """Get the options flow for this handler."""
+        return OptionsFlow(config_entry)
+
+
+class OptionsFlow(config_entries.OptionsFlowWithConfigEntry):
+    """Option flow handler."""
+
+    def _async_abort_entries_match(
+        self, user_input: dict[str, Any] | None
+    ) -> dict[str, str]:
+        """Validate the user input against other config entries."""
+        if user_input is None:
+            return {}
+
+        errors: dict[str, str] = {}
+        for entry in [
+            entry
+            for entry in self.hass.config_entries.async_entries(DOMAIN)
+            if entry is not self.config_entry
+        ]:
+            if all(
+                item in entry.data.items()
+                for item in user_input.items()
+                if item[0] in KEY_OPTIONS
```

Comment:
Should key options not also contain host/username?

Reply:
May be you want to add the save search and and folder but for a different server, may be it would be good to include server too

Reply:
But just for the check, not as an option to be set...

Reply:
Yes, we should allow same search and folder but different server.